runtime/compiler/trj9/control/HookedByTheJit.cpp

```diff
@@ -564,7 +564,11 @@ static void jitHookInitializeSendTarget(J9HookInterface * * hook, UDATA eventNum
          } // if (TR::Options::sharedClassCache())
       if (count == -1) // count didn't change yet
          {
-         count = getCount(romMethod, optionsJIT, optionsAOT);
+         if (!TR::Options::getCountsAreProvidedByUser() &&
+            fe->isClassLibraryMethod((TR_OpaqueMethodBlock *)method))
```

Comment:
The description for this change says we want to target "methods belonging to bootstrap classes that are executed a lot during start-up but to a lesser degree during steady state" but the check here is `isClassLibraryMethod()` which is basically every method from JCL and J9VM classes (regardless of when and how often it's executed) isn't it? Is there some other logic elsewhere that actually limits the behaviour to what was desired?

Reply:
What I meant in the description was that in general methods belonging to bootstrap classes are executed a lot during start-up/JVM-bootstrap by their nature. The counts are set when the classes/methods are loaded, so there is no way to dynamically determine how often they are going to be executed at that point.

Reply:
Is there any concern that this could negatively affect programs that make heavy use of the JCL long after startup? E.g. programs that use `java/{util/math/io}` or something of that nature in their primary task? Since the functionality will be disabled until more performance measurements are done I don't think we need an answer to this now, but finding/measuring important programs that go against this assumption is probably a worthwhile data point.

Reply:
My first attempt was actually using this kind of filtering thinking that throughput in Daytrader/AcmeAir would be affected. To my surprise it wasn't (actually I got a small improvement which may or may not be noise). The goal of this commit is to have a build that I could use for more benchmarks and on other platforms using the automation provided by the perf monitoring infrastructure.